AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/deployments/DeploymentDtoCsvBodyWriter.java

```diff
@@ -0,0 +1,162 @@
+package ch.mobi.itc.mobiliar.rest.deployments;
+
+import ch.mobi.itc.mobiliar.rest.dtos.AppWithVersionDTO;
+import ch.mobi.itc.mobiliar.rest.dtos.DeploymentDTO;
+import ch.mobi.itc.mobiliar.rest.dtos.DeploymentParameterDTO;
+import ch.puzzle.itc.mobiliar.common.util.ConfigurationService;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+@Provider
+@Produces({"text/comma-separated-values", "text/csv"})
+public class DeploymentDtoCsvBodyWriter implements MessageBodyWriter<List<DeploymentDTO>>{
+
+    private static final String CSV_SEPARATOR = ConfigurationService.getProperty(ConfigurationService.ConfigKey.CSV_SEPARATOR);
+
+    @Override
+    public boolean isWriteable(Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType) {
+        // Ensure that we're handling only List<DeploymentDTO> objects.
+        boolean isWritable = false;
+        if (List.class.isAssignableFrom(aClass) && type instanceof ParameterizedType) {
+            ParameterizedType parameterizedType = (ParameterizedType) type;
+            Type[] actualTypeArgs = (parameterizedType.getActualTypeArguments());
+            isWritable = (actualTypeArgs.length == 1 && actualTypeArgs[0].equals(DeploymentDTO.class));
+        }
+        return isWritable;
+    }
+
+    @Override
+    public long getSize(List<DeploymentDTO> deploymentDTOS, Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType) {
+        //used for contentLength header, -1 = calculate it
+        return -1;
+    }
+
+    @Override
+    public void writeTo(List<DeploymentDTO> deploymentDTOS, Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, Object> httpHeaders, OutputStream outputStream) throws WebApplicationException {
```

Comment:
I believe this duplicates code in the JSF GUI. Please add a comment to both classes that contain the code.

Reply:
I didn't realise the JSF CSV export code was still there. It's some howsame but different.

Reply:
@yvespp Yes, I had to create an additional endpoint: /deployments/filterCsvExport
because the Angular-UI needs the total count number from the header information for the pagination. So I can not simply return a List<DeploymentDTO>. And Swagger does not support endpoints with identical path and different return type see: swagger-api/swagger-core#935